BUILD.gn

```diff
@@ -0,0 +1,530 @@
+
+config("xwalk_runtime_config") {
+  # TODO(rakuco): Remove this once we stop supporting Ubuntu Precise or
+  # default to use_sysroot==1. This is only required because Precise's dbus
+  # 1.4.18 causes
+  # <https://bugs.chromium.org/p/chromium/issues/detail?id=263960#c6>.
+  if (is_clang && !use_sysroot) {
+    cflags = [ "-Wno-reserved-user-defined-literal" ]
+  }
+}
+
+source_set("xwalk_runtime") {
+  public_configs = [ "//build/config:feature_flags" ]
+  configs += [
+    # TODO(jschuh): crbug.com/167187 fix size_t to int truncations.
+    "//build/config/compiler:no_size_t_to_int_warning",
+    ":xwalk_runtime_config",
+  ]
+
+  defines = [
+    "XWALK_VERSION=\"$xwalk_version\"",
+    "CHROME_VERSION=\"$chrome_version\"",
+  ]
+
+  deps = [
+    ":generate_upstream_blink_version",
+    "//base:base",
+    "//base:base_static",
+    "//base:i18n",
+    "//base/third_party/dynamic_annotations",
+    "//cc",
+    "//components/autofill/content/browser",
+    "//components/autofill/content/renderer",
+    "//components/autofill/core/browser",
+    "//components/cdm/renderer",
+    "//components/devtools_http_handler",
+    "//components/resources:components_resources",
+    "//components/strings",
+    "//components/url_formatter",
+    "//components/user_prefs",
+    "//components/visitedlink/browser",
+    "//components/visitedlink/renderer",
+    "//content:content",
+    "//content/public/app:both",
+    "//content/public/browser",
+    "//content/public/child",
+    "//content/public/common",
+    "//content/public/utility",
+    "//gin:gin",
+    "//ipc:ipc",
+    "//media:media",
+    "//net:net",
+    "//net:net_resources",
+    "//skia:skia",
+    "//storage/browser:browser",
+    "//storage/common",
+    "//third_party/WebKit/public:blink",
+    "//third_party/boringssl:boringssl",
+    "//ui/base",
+    "//ui/gl",
+    "//ui/shell_dialogs",
+    "//ui/snapshot",
+    "//url:url",
+    "//v8",
+    "//xwalk/application:xwalk_application_lib",
+    "//xwalk/experimental:experimental",
+    "//xwalk/extensions:xwalk_extensions",
+    "//xwalk/resources:xwalk_pak",
+    "//xwalk/resources:xwalk_resources",
+    "//xwalk/resources:xwalk_resources_100_percent_pak",
+    "//xwalk/resources:xwalk_resources_200_percent_pak",
+    "//xwalk/resources:xwalk_resources_300_percent_pak",
+    "//xwalk/sysapps:sysapps",
+  ]
+  if (is_linux) {
+    deps += [ "//xwalk/resources:xwalk_strings" ]
+  }
+
+  include_dirs = [
+    "..",
+    "$root_gen_dir",
+    "$root_gen_dir/xwalk",
+  ]
+
+  sources = [
+    "//extensions/common/constants.cc",
+    "//extensions/common/constants.h",
+    "//extensions/common/url_pattern.cc",
+    "//extensions/common/url_pattern.h",
+    "runtime/app/android/xwalk_main_delegate_android.cc",
+    "runtime/app/android/xwalk_main_delegate_android.h",
+    "runtime/app/xwalk_main_delegate.cc",
+    "runtime/app/xwalk_main_delegate.h",
+    "runtime/browser/android/cookie_manager.cc",
```

Comment:
I did consider to remove those Android/Windows files, but finally just leave them alone.  The reason is that in //build/config/BUILDCONFIG.gn,  there is a "sources_assignment_filter",  it will filter out all platform-specific files automatically. To avoid mistakes, I just keep all the source files here.  
If you think it is not good, I can remove them in this linux PR. But I think we'll add them back sooner or later.  so could we just keep the them?